src/librustc/ich/impls_syntax.rs

```diff
@@ -74,7 +74,10 @@ impl<'a> HashStable<StableHashingContext<'a>> for SourceFile {
         (name_hash as u64).hash_stable(hcx, hasher);
         name_was_remapped.hash_stable(hcx, hasher);
 
-        src_hash.hash_stable(hcx, hasher);
+        src_hash.kind.hash_stable(hcx, hasher);
+        for byte in src_hash.as_bytes() {
+            byte.hash_stable(hcx, hasher);
+        }
```

Comment:
`src_hash` itself should implement `HashStable` (you can even try to add `#[derive(HashStable_Generic)]` on the definition, I forget which cases it works in).

Reply:
HashStable_Generic seemed to work here. Thanks!